src/Molecules/ToastMessage/ToastMessage.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useImperativeHandle, useState } from 'react';
+import { AnimatePresence, motion } from 'framer-motion';
+
+import { Flexbox } from '../..';
+import { BoxContainer, StyledTypography } from './ToastMessage.styled';
+import { Props } from './ToastMessage.types';
+
+const ToastMessages: React.FC<Props> = React.forwardRef(
+  ({ label, icon, linkButton, isVisible, timeout = 6000 }, ref) => {
+    const [visible, setVisibility] = useState<boolean>(isVisible || false);
+    const [mouseEnter, setMouseEnter] = useState<boolean>(false);
+
+    const duration = 0.35;
+    const variants = {
+      initial: {
+        opacity: 0,
+      },
+      enter: {
+        opacity: 1,
+        transition: {
+          duration,
+          ease: 'easeInOut',
+        },
+      },
+      exit: {
+        opacity: 0,
+        transition: { duration },
+      },
+    };
+
+    useImperativeHandle(ref, () => ({
+      setVisible: () => {
+        setVisibility(!visible);
+      },
+    }));
+
+    useEffect(() => {
+      if (typeof ref === 'undefined' || ref === null) {
+        setVisibility(isVisible || false);
+      }
+
+      if (mouseEnter) {
+        setVisibility(true);
+      }
+    }, [isVisible, ref, mouseEnter]);
+
+    useEffect(() => {
+      const timer = setTimeout(() => {
+        if (visible === true && !mouseEnter) {
```

Comment:
no need to explicitly compare with `true` since `visible` is a boolean
```suggestion
        if (visible && !mouseEnter) {
```